tests/v2/validation/pipeline/airgapsetup/Jenkinsfile.airgap.setup

```diff
@@ -0,0 +1,125 @@
+                        userRemoteConfigs: repo
+                      ])
+            }
+        stage('Configure and Build') {
+          config = config.replace('${AWS_SECRET_ACCESS_KEY}', env.AWS_SECRET_ACCESS_KEY)
+          config = config.replace('${AWS_ACCESS_KEY_ID}', env.AWS_ACCESS_KEY_ID)
+
+          writeFile file: 'config.yml', text: config
+
+          def decoded = new String(env.AWS_SSH_PEM_KEY.decodeBase64())
+          writeFile file: 'key.pem', text: decoded
+          
+          env.CATTLE_TEST_CONFIG=rootPath+'config.yml'
+
+          sh "./configure.sh"
+          sh "./build.sh"
+        }
+        stage('Setup'){
+            sh returnStdout: true, script: 'wget -qO ./yq https://github.com/mikefarah/yq/releases/latest/download/yq_linux_amd64'
+            sh returnStdout:true, script: 'chmod a+x ./yq'
+
+            println "AWS s3 config ${s3Config}"
+            
+            AWS_S3_BUCKET = sh (
+                  script: "echo '${s3Config}' | ./yq '.s3Bucket'",
+                  returnStdout: true
+                ).trim()
+                
+            println "AWS_S3_BUCKET: ${AWS_S3_BUCKET}"
+
+            AWS_S3_REGION =    sh (
+                  script: "echo '${s3Config}' | ./yq '.s3BucketRegion'",
+                  returnStdout: true
+                ).trim()
+            println "AWS_S3_REGION: ${AWS_S3_REGION}"
+
+        }
+        stage('Run Module Test') {
+          try {
+            sh """
+              docker run --name ${testContainer} -t -v ${homePath}key.pem:${rootPath}key.pem --env-file ${envFile} ${imageName} sh -c "
+              /root/go/bin/gotestsum --format standard-verbose --packages=${testsDir} --junitfile ${testResultsOut} --jsonfile ${testResultsJSON} -- -timeout=${timeout} -v ${params.TEST_CASE};
+              if [ -f ${rootPath}reporter ]; then ${rootPath}reporter; fi"
+            """
+          } catch(err) {
+              echo 'Test run had failures. Collecting results...'
+          }
+        }
+        stage('Upload tf files to S3 bucket'){
+          try{
+              sh "docker cp ${testContainer}:/root/ ./"
+              sh "ls -lR"
+              script {
+                sh """
+                    if ls $TERRAFORM_DIR/. 1> /dev/null 2>&1; then
+                        echo "Terraform files found, starting upload."
+                    else
+                        echo "No Terraform files found!"
+                        exit 1
+                    fi
+                      docker run --rm \
+                        -e AWS_ACCESS_KEY_ID=$AWS_ACCESS_KEY_ID \
+                        -e AWS_SECRET_ACCESS_KEY=$AWS_SECRET_ACCESS_KEY \
+                        -e AWS_DEFAULT_REGION=$AWS_S3_REGION \
+                        -v $TERRAFORM_DIR/:/root/terraform/ \
+                        amazon/aws-cli \
+                        s3 cp /root/terraform/. s3://$AWS_S3_BUCKET/terraform/ --recursive
```

Comment:
to Max's point, this may need a path configured to avoid overwriting other people's jobs.